### ISSUE TITLE
Marshalling time of custom types is dominated by hash computation

### ISSUE THREAD

**Issue by Imberflur** [mun-lang/mun#224]
[flamegraph.svg.zip](https://github.com/mun-lang/mun/files/4774178/flamegraph.svg.zip)
![image](https://user-images.githubusercontent.com/2002109/84561785-55997a80-ad1d-11ea-851f-c1a3c3a85dbc.png)

This seems like something that could be vastly improved by caching the `type_guid` computation for `StructRef` but there may be a reason that this is not being done.

```rs
impl ReturnTypeReflection for StructRef {
    type Marshalled = RawStruct;

    fn type_name() -> &'static str {
        "struct"
    }
}
```

**Comment by Wodann** [comment]
Good find. There is no reason why `ReturnTypeReflection::<StructRef>::type_guid` can not be constant. I'll look into optimising this, thanks! :slightly_smiling_face:

Also, exciting to see that you are considering this for Veloren :+1:

**Comment by baszalmstra** [comment]
@Wodann You are currently working on a fix for this right?

**Comment by Wodann** [comment]
I have a branch with the improvement, but wanted to add some additional benchmarks as well. I was working on a second performance improvement based on that. 

The first pass can be found at: https://github.com/Wodann/mun/tree/improve/struct-marshalling-perf

**Comment by Wodann** [comment]
Resolved by #238.
